projects/cardiovascular-patient-app/src/app/last-report-bp/last-report-bp.component.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { LastReportBPComponent } from './last-report-bp.component';
+import { DataLayerManagerService, ManagedDataLayer } from 'ngx-charts-on-fhir';
+import { BehaviorSubject, EMPTY } from 'rxjs';
+import { LastReportBPModule } from './last-report-bp.module';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { MatCardHarness } from '@angular/material/card/testing';
+
+
+class MockLayerManager {
+  allLayers$ = new BehaviorSubject<ManagedDataLayer[]>([]);
+  selectedLayers$ = EMPTY;
+  availableLayers$ = EMPTY;
+}
+
+describe('LastReportBPComponent', () => {
+  let component: LastReportBPComponent;
+  let fixture: ComponentFixture<LastReportBPComponent>;
+  let layerManager: MockLayerManager;
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    layerManager = new MockLayerManager();
+    await TestBed.configureTestingModule({
+      imports: [LastReportBPModule, NoopAnimationsModule],
+      providers: [
+        { provide: DataLayerManagerService, useValue: layerManager }
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(LastReportBPComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should extract the last data point of each dataset in the "Blood Pressure" layer', async () => {
+    const layers: ManagedDataLayer[] = [
+      {
+        id: '1',
+        name: 'Blood Pressure',
+        category: ['C', 'D'],
+        datasets: [
+          {
+            data: [
+              { x: 1609175027000, y: 120 },
+              { x: 1609175027000, y: 122 },
+            ],
+            label: "Diastolic Blood Pressure"
+          },
+          {
+            data: [
+              { x: 1676536294000, y: 80 },
+              { x: 1676536294000, y: 78 },
+            ],
+            label: "Systolic Blood Pressure"
+          },
+        ],
+        scale: { id: '1' },
+      },
+    ];
+    layerManager.allLayers$.next(layers);
+
+    const cardHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, MatCardHarness);
+    const contentText = await cardHarness.getText();
+    expect(contentText).toContain('Last BP reported was 78/122 on 16 Feb 2023 at 8:31 AM');
```

Comment:
This test is failing when I run it locally. Looks like a timezone issue. Tests should not depend on system time being set to a specific timezone.
```
Chrome Headless 110.0.5481.96 (Linux x86_64) LastReportBPComponent should extract the last data point of each dataset in the "Blood Pressure" layer FAILED
        Expected 'Last BP reported was 78/122 on 16 Feb 2023 at 3:31 AM' to contain 'Last BP reported was 78/122 on 16 Feb 2023 at 8:31 AM'.
            at <Jasmine>
            at UserContext.<anonymous> (projects/cardiovascular-patient-app/src/app/last-report-bp/last-report-bp.component.spec.ts:71:25)
            at Generator.next (<anonymous>)
            at asyncGeneratorStep (node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js:3:1)
            at _next (node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js:22:1)
```